### PR TITLE
Integrate conn_graph_facts into devutils

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)
This PR updates ```devutils``` and ```conn_graph_facts.py```.

1. Use conn_graph_facts to retrieve pdu info and console info in devutils
2. Update conn_graph_facts library to support standlone running.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to update ```devutils``` and ```conn_graph_facts.py```.

#### How did you do it?
1. Update ```conn_graph_facts.py``` to support calling in ```devutils```
2. Update ```devutils``` to get pdu and console info from ```conn_graph_facts```. Info in inventory is used as a fallback.

#### How did you verify/test it?
1. Verified pdu info retrieved from ```conn_graph_facts``` is correct
2. Verified console info retrieved from ```conn_graph_facts``` is correct
3. Verify ```conn_graph_facts.py``` is compatible with Ansible after update.

#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
No.

### CODE REVIEW

File: ansible/devutils [hunk 1]
```diff
@@ -186,7 +249,10 @@ def action_pdu(parameters, action):
         if parameters['json']:
             data.append(status)
         else:
-            data.append([status[x] for x in header])
+            try:
+                data.append([status[x] for x in header])
+            except TypeError:
+                pass
```
Comment (yxieca) [inline ansible/devutils:255]: In my experience, if we got here, it is usually indicating problem with pdu manager/controller. We should surface the problem for debug/fix.

File: ansible/library/conn_graph_facts.py [hunk 2]
```diff
@@ -325,8 +332,16 @@ def get_host_connections(self, hostname):
         """
         return self.links.get(hostname)
 
-    def contains_hosts(self, hostnames):
-        return set(hostnames) <= set(self.devices)
+    def contains_hosts(self, hostnames, part):
+        if not part:
+            return set(hostnames) <= set(self.devices)
+        # It's possible that not all devices are found in connect_graph when using in devutil
+        THRESHOLD = 0.8
```
Comment (yxieca) [inline ansible/library/conn_graph_facts.py:339]: Wondering what 80% found help? Maybe we could explicitly callout all found / partial / none found?

File: ansible/devutils (hunk 1 above)
Comment (bingwang-ms) [reply ansible/devutils:255]: Indeed. The excception is usually raised when some error happens in target function. I will cancel the ```try-except``` here.

File: ansible/library/conn_graph_facts.py (hunk 2 above)
Comment (bingwang-ms) [reply ansible/library/conn_graph_facts.py:339]: The reason for I introduced partial found is not all devices defined in ```inventory``` are listed in ```conn_graph```. The default logic in ```find_graph``` in ```conn_graph_facts.py``` is attempting to find a ```xml``` file which contains all devices. So the logic will not work in ```devutils```. So I select the figure ```80%```, which means ```most``` devices are found.

File: ansible/devutils [hunk 3]
```diff
@@ -187,7 +250,7 @@ def action_pdu(parameters, action):
             data.append(status)
         else:
             data.append([status[x] for x in header])
-
+
```
Comment (yxieca) [inline ansible/devutils:253]: Can you remove this trailing spaces?

**Review by yxieca — APPROVED** [review]
Please remove the trailing spaces before check in.

File: ansible/devutils (hunk 3 above)
Comment (bingwang-ms) [reply ansible/devutils]: Sure. Updated. Thanks

**Pull request merged**